### PR TITLE
Fixes #9934: Clean tags data model 

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/9934

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-core/src/main/scala/com/normation/rudder/domain/policies/Tags.scala [hunk 1]
```diff
@@ -44,11 +44,15 @@ import com.normation.utils.Control._
  * Tags that apply on Rules and Directives
  * We do not warranty unicity of tags name, only of tagsname + tagvalue
  */
-final case class TagName( name : String )
 
-final case class TagValue( value : String )
+object Tag {
+  final implicit class TagName ( val value : String )
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/domain/policies/Tags.scala]: I think we really need them to be implicit *case* class, else the equals/hashcode won't be what we want at all

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
@fanf case class back, going back to implicits;

**Comment by VinceMacBuche** [comment]
PR rebased

**Pull request merged**